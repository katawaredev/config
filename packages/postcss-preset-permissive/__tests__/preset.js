const fs = require("fs");
const { promisify } = require("util");
const path = require("path");
const postcss = require("postcss");
const preset = require("../src");

const readFileAsync = promisify(fs.readFile);

test("handles css correctly", async () => {
  const cssFile = path.join(__dirname, "app.css");
  const source = await readFileAsync(cssFile);
  const processor = postcss([preset]);
  const { css } = await processor.process(source, {
    from: cssFile,
    to: cssFile,
  });
  expect(css).toMatchSnapshot();
});
