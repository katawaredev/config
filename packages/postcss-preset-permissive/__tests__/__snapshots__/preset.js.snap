// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles css correctly 1`] = `
"@import url(\\"https://fonts.googleapis.com/css2?family=Roboto&display=swap\\");
/* http://meyerweb.com/eric/tools/css/reset/
   v5.0.1 | 20191019
   License: none (public domain)
*/
html, body, div, span, applet, object, iframe,
h1, h2, h3, h4, h5, h6, p, blockquote, pre,
a, abbr, acronym, address, big, cite, code,
del, dfn, em, img, ins, kbd, q, s, samp,
small, strike, strong, sub, sup, tt, var,
b, u, i, center,
dl, dt, dd, menu, ol, ul, li,
fieldset, form, label, legend,
table, caption, tbody, tfoot, thead, tr, th, td,
article, aside, canvas, details, embed,
figure, figcaption, footer, header, hgroup,
main, menu, nav, output, ruby, section, summary,
time, mark, audio, video {
	margin: 0;
	padding: 0;
	border: 0;
	font-size: 100%;
	font: inherit;
	vertical-align: baseline;
}
/* HTML5 display-role reset for older browsers */
article, aside, details, figcaption, figure,
footer, header, hgroup, main, menu, nav, section {
	display: block;
}
/* HTML5 hidden-attribute fix for newer browsers */
*[hidden] {
    display: none;
}
body {
	line-height: 1;
}
menu, ol, ul {
	list-style: none;
}
blockquote, q {
	quotes: none;
}
blockquote:before, blockquote:after,
q:before, q:after {
	content: '';
	content: none;
}
table {
	border-collapse: collapse;
	border-spacing: 0;
}
a {
  all: initial;
}
@custom-media --viewport-medium (max-width: 50rem);
@custom-selector :--heading h1, h2, h3, h4, h5, h6;
:root {
  --mainColor: #12345678;
}
body {
  color: var(--mainColor);
  font-family: system-ui;
  overflow-wrap: break-word;
}
:--heading {
  /* stylelint-disable-next-line plugin/no-unsupported-browser-features */
  background-image: -webkit-image-set(
    url(img/heading.png) 1x,
    url(img/heading@2x.png) 2x
  );
  background-image: image-set(
    url(img/heading.png) 1x,
    url(img/heading@2x.png) 2x
  );

  @media (--viewport-medium) {
    margin-block: 0;
  }
}
a {
  color: rgb(0 0 100% / 90%);

  &:hover,
  &:focus {
    color: rebeccapurple;
  }
}
"
`;
